backend/cmd/start.go

```diff
@@ -242,19 +286,106 @@ func newStartCommand() *cobra.Command {
 	cmd.Flags().Bool(flagNoEmbedEtcd, viper.GetBool(flagNoEmbedEtcd), "don't embed etcd, use external etcd instead")
+	_ = cmd.Flags().SetAnnotation(flagNoEmbedEtcd, "categories", []string{"store"})
+
+	// Etcd TLS flags
+	cmd.Flags().String(flagEtcdCertFile, viper.GetString(flagEtcdCertFile), "path to the client server TLS cert file")
+	_ = cmd.Flags().SetAnnotation(flagEtcdCertFile, "categories", []string{"store"})
+	cmd.Flags().String(flagEtcdKeyFile, viper.GetString(flagEtcdKeyFile), "path to the client server TLS key file")
+	_ = cmd.Flags().SetAnnotation(flagEtcdKeyFile, "categories", []string{"store"})
+	cmd.Flags().Bool(flagEtcdClientCertAuth, viper.GetBool(flagEtcdClientCertAuth), "enable client cert authentication")
+	_ = cmd.Flags().SetAnnotation(flagEtcdClientCertAuth, "categories", []string{"store"})
+	cmd.Flags().String(flagEtcdTrustedCAFile, viper.GetString(flagEtcdTrustedCAFile), "path to the client server TLS trusted CA cert file")
+	_ = cmd.Flags().SetAnnotation(flagEtcdTrustedCAFile, "categories", []string{"store"})
+	cmd.Flags().String(flagEtcdPeerCertFile, viper.GetString(flagEtcdPeerCertFile), "path to the peer server TLS cert file")
+	_ = cmd.Flags().SetAnnotation(flagEtcdPeerCertFile, "categories", []string{"store"})
+	cmd.Flags().String(flagEtcdPeerKeyFile, viper.GetString(flagEtcdPeerKeyFile), "path to the peer server TLS key file")
+	_ = cmd.Flags().SetAnnotation(flagEtcdPeerKeyFile, "categories", []string{"store"})
+	cmd.Flags().Bool(flagEtcdPeerClientCertAuth, viper.GetBool(flagEtcdPeerClientCertAuth), "enable peer client cert authentication")
+	_ = cmd.Flags().SetAnnotation(flagEtcdPeerClientCertAuth, "categories", []string{"store"})
+	cmd.Flags().String(flagEtcdPeerTrustedCAFile, viper.GetString(flagEtcdPeerTrustedCAFile), "path to the peer server TLS trusted CA file")
+	_ = cmd.Flags().SetAnnotation(flagEtcdPeerTrustedCAFile, "categories", []string{"store"})
+
+	// Mark the old etcd flags as deprecated and maintain backward compability
+	cmd.Flags().SetNormalizeFunc(aliasNormalizeFunc)
 
 	// Load the configuration file but only error out if flagConfigFile is used
 	if err := viper.ReadInConfig(); err != nil && configFile != "" {
 		setupErr = err
 	}
 
+	// Use our custom template for the start command
+	cobra.AddTemplateFunc("categoryFlags", categoryFlags)
+	cmd.SetUsageTemplate(startUsageTemplate)
+
 	return cmd
 }
+
+func categoryFlags(category string, flags *pflag.FlagSet) *pflag.FlagSet {
+	flagSet := pflag.NewFlagSet(category, pflag.ContinueOnError)
+
+	flags.VisitAll(func(flag *pflag.Flag) {
+		if categories, ok := flag.Annotations["categories"]; ok {
+			if stringsutil.InArray(category, categories) {
+				flagSet.AddFlag(flag)
+			}
+		} else if category == "" {
+			// If no category was specified, return all flags without a category
+			flagSet.AddFlag(flag)
+		}
+	})
+
+	return flagSet
+}
+
+func aliasNormalizeFunc(f *pflag.FlagSet, name string) pflag.NormalizedName {
```

Comment:
Thanks for doing this! Supporting deprecated flags is important in the interim 👍